lib/graphql_devise/rails/routes.rb

```diff
@@ -1,111 +1,98 @@
-        else
-          new_mutation = Class.new(mutation)
-          new_mutation.graphql_name("#{resource}#{action.to_s.camelize(:upper)}")
-          new_mutation.field(:authenticatable, authenticatable_type, null: true)
+      prepared_queries = GraphqlDevise::QueriesPreparer.call(
+        resource:             resource,
+        queries:              GraphqlDevise::OperationSanitizer.call(
+          default: GraphqlDevise::QueriesPreparer::DEFAULT_QUERIES, **param_operations
+        ),
+        authenticatable_type: authenticatable_type
+      )
 
-          new_mutation
-        end
-        used_mutation.instance_variable_set(:@resource_name, mapping_name)
+      add_gql_devise_mutations!(prepared_mutations, additional_mutations)
+      add_gql_devise_queries!(prepared_queries, additional_queries)
 
-        GraphqlDevise::Types::MutationType.field("#{mapping_name}_#{action}", mutation: used_mutation)
-      end
-      additional_mutations.each do |action, mutation|
-        GraphqlDevise::Types::MutationType.field(action, mutation: mutation)
-      end
+      Devise.mailer.helper(GraphqlDevise::MailerHelper)
 
-      if (used_mutations.present? || additional_mutations.present?) &&
-         (Gem::Version.new(GraphQL::VERSION) <= Gem::Version.new('1.10.0') || GraphqlDevise::Schema.mutation.nil?)
-        GraphqlDevise::Schema.mutation(GraphqlDevise::Types::MutationType)
+      devise_scope mapping_name do
+        post path, to: 'graphql_devise/graphql#auth'
+        get  path, to: 'graphql_devise/graphql#auth'
       end
+    end
 
-      used_queries = if only_operations.present?
-        default_queries.slice(*only_operations)
-      else
-        default_queries.except(*skipped_operations)
-      end
-      used_queries.each do |action, query|
-        used_query = if custom_operations[action].present?
-          custom_operations[action]
-        else
-          new_query = Class.new(query)
-          new_query.graphql_name("#{resource}#{action.to_s.camelize(:upper)}")
-          new_query.type(authenticatable_type, null: true)
+    private
+
+    def validate_gql_devise_operations!(param_operations)
+      GraphqlDevise::OperationChecker.call(
+        mutations: GraphqlDevise::MutationsPreparer::DEFAULT_MUTATIONS,
+        queries:   GraphqlDevise::QueriesPreparer::DEFAULT_QUERIES,
+        **param_operations
+      )
+    end
 
-          new_query
-        end
-        used_query.instance_variable_set(:@resource_name, mapping_name)
+    def add_gql_devise_mutations!(prepared, additional)
+      all_mutations = prepared.merge(additional)
 
-        GraphqlDevise::Types::QueryType.field("#{mapping_name}_#{action}", resolver: used_query)
+      all_mutations.each do |action, mutation|
+        GraphqlDevise::Types::MutationType.field(action, mutation: mutation)
       end
-      additional_queries.each do |action, resolver|
+
+      # if all_mutations.present? && GraphqlDevise::Schema.try(:mutation).nil?
```

Comment:
I don't think try should replace the version check. I think all versions respond to the method actually. Plus the mutation method behaves differently on gql 1.10